addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -328,6 +328,125 @@ options.registry.sizing_x = options.registry.sizing.extend({
     },
 });
 
+options.registry.layout_column = options.Class.extend({
+    /**
+     * @override
+     */
+    start: function () {
+        this.$overlay.find('.o_handle.o_add_column, .o_handle.o_remove_column')
+            .removeClass('readonly')
+            .on('click', this._onButtonClick.bind(this));
+        return this._super.apply(this, arguments);
+    },
+    /**
+     * @override
+     */
+    onFocus: function () {
+        this._toggleButtons();
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Add new column which is clone of last column
+     *
+     * @private
+     */
+    _addColumn: function () {
+        this.trigger_up('request_history_undo_record', {$target: this.$target});
+        var $lastColumn = this._getColumns().last();
+        $lastColumn.clone().insertAfter($lastColumn);
+        this._updateSnippet();
+    },
+    /**
+     * Return all the columns of currently selected snippet.
+     *
+     * @private
+     * @returns {jQuery}
+     */
+    _getColumns: function () {
+        return this.$target.find('.container > .row > [class*="col-md-"]');
```

Comment:
Would be a simple `this.$target.children()` thanks to `data-target` mentioned below.